crates/steel-parser/src/lexer.rs

```diff
@@ -644,13 +655,27 @@ pub enum TokenError {
     IncompleteComment,
     InvalidEscape,
     InvalidCharacter,
-    MalformedRational,
-    MalformedHexInteger,
-    MalformedOctalInteger,
-    MalformedBinaryInteger,
+    ZeroDenominator,
     MalformedByteEscape,
 }
 
+impl std::fmt::Display for TokenError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TokenError::UnexpectedChar(_) => write!(f, "unexpected char"),
+            TokenError::IncompleteString => write!(f, "incomplete string"),
+            TokenError::IncompleteIdentifier => write!(f, "incomplete identifier"),
+            TokenError::IncompleteComment => write!(f, "incomplete comment"),
+            TokenError::InvalidEscape => write!(f, "invalid escape"),
+            TokenError::InvalidCharacter => write!(f, "invalid character"),
+            TokenError::ZeroDenominator => {
+                write!(f, "division by zero is not allowed in rational literals")
+            }
+            TokenError::MalformedByteEscape => write!(f, "malformed byte escape"),
```

Comment:
I plan to revisit all token and parser errors to make them as useful as possible.